crates/primitives/src/header.rs

```diff
@@ -423,6 +423,12 @@ impl Encodable for Header {
         self.mix_hash.encode(out); // Encode mix hash.
         B64::new(self.nonce.to_be_bytes()).encode(out); // Encode nonce.
 
+        // The following code is needed only to handle proptest-generated headers that are
+        // technically invalid.
+        //
+        // TODO: make proptest generate more valid headers, ie if there is no base fee, there
+        // should be no withdrawals root or any future fork field.
```

Comment:
can we open an issue for manual arbitrary impl?

Reply:
yes https://github.com/paradigmxyz/reth/issues/6329